medusa/post_processor.py

```diff
@@ -287,6 +288,16 @@ def list_associated_files(self, file_path, base_name_only=False, subtitles_only=
 
         return file_path_list
 
+    @staticmethod
+    def rar_basename(filepath, filelist):
+        videofile = os.path.basename(filepath)
+        rars = (x for x in filelist if rarfile.is_rarfile(x))
+
+        for rar in rars:
+            content = rarfile.RarFile(rar).namelist()
```

Comment:
As i said `process_media(path, videoInRar, nzbName, process_method, force, is_priority, ignore_subs, result)`  passed videoInRar, and `process_media` calls list_associated_files. so you don't need to call RAR again to check rar contents